pkgs/development/python-modules/scan-build/default.nix

```diff
@@ -0,0 +1,21 @@
+{ lib, buildPythonPackage, fetchPypi, isPy27 }:
+
+buildPythonPackage rec {
+  pname = "scan-build";
+  version = "2.0.18";
+  disabled = isPy27;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "296bb899dc4c126b985fca0c15c9ec20cf758b7c7ff830db64c4ca971d886797";
+  };
+
+  doCheck = false;
```

Comment:
please checkout from github and run unit tests https://github.com/rizsotto/scan-build/tree/master/tests/unit

Reply:
important for testing regressions

Reply:
Alrighty, so I've done this by changing substituting `doCheck = false;` with:

```
  checkInputs = [ nose ];
  checkPhase = "nosetests tests/unit";
```

However, now I'm getting a test failure, and I'm not sure what I need to do to get it to pass. This is the output of the failed test:

```
..........................F...............................S..S.S.S.S.....S.S
======================================================================
FAIL: test_get_clang_arguments_fails (test_clang.ClangGetArgumentsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/private/tmp/nix-build-python3.7-scan-build-2.0.18.drv-0/source/tests/unit/test_clang.py", line 38, in test_get_clang_arguments_fails
    sut.get_arguments(['clang', '-x', 'c', 'notexist.c'], '.')
AssertionError: Exception not raised
-------------------- >> begin captured logging << --------------------
root: DEBUG: exec command ['clang', '-###', '-x', 'c', 'notexist.c'] in /private/tmp/nix-build-python3.7-scan-build-2.0.18.drv-0/source
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 76 tests in 1.768s

FAILED (SKIP=7, failures=1)
```

I've checked simple things like check that `clang` is in the build environment. Not sure where to go from here.

Reply:
see my other comment.

clang, if listed as a build input, will not be present during checkPhase